internal/cmd/gocdk/internal/docker/docker.go

```diff
@@ -0,0 +1,211 @@
+// Copyright 2019 The Go Cloud Development Kit Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package docker provides a client for interacting with the local Docker
+// daemon. This currently shells out to the Docker CLI, but could use the HTTP
+// API directly in the future.
+package docker
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"os/exec"
+	"strings"
+	"time"
+
+	"golang.org/x/xerrors"
+)
+
+// Client is a client for a Docker daemon. The zero value is a Docker client
+// that uses the current process's environment. Methods on Client are safe to
+// call concurrently from multiple goroutines.
+type Client struct {
+	env []string
+}
+
+// New creates a new client with the given environment variables.
+// Calling docker.New(nil) is the same as new(docker.Client) and uses the
+// current process's environment.
+func New(env []string) *Client {
+	return &Client{env: env}
+}
+
+// Image stores metadata about a Docker image.
+type Image struct {
+	ID         string
+	Repository string
+	Tag        string
+	Digest     string
+	CreatedAt  time.Time
+}
+
+// ListImages list images the Docker daemon has stored locally that match filterRef.
+func (c *Client) ListImages(ctx context.Context, filterRef string) ([]Image, error) {
```

Comment:
`*Image`? Not sure there's any advantage to copying.

Reply:
Done.